File: System/Workspace/Typescript/libs/workspace/system/src/lib/meta/IRelationType.ts
import { Multiplicity } from '../Multiplicity';
import { IAssociationType } from './IAssociationType';
import { IMetaObject } from './IMetaObject';
import { IRoleType } from './IRoleType';

export interface IRelationType extends IMetaObject {
  associationType: IAssociationType;

  roleType: IRoleType;

  multiplicity: Multiplicity;

  isDerived: boolean;
}

File: dotnet/System/Workspace/Allors.Workspace.Meta/IRelationType.cs
// <copyright file="IRelationType.cs" company="Allors bv">
// Copyright (c) Allors bv. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>
// <summary>Defines the RelationType type.</summary>

namespace Allors.Workspace.Meta
{
    /// <summary>
    /// A relation type defines the state and behavior for
    /// a set of association types and role types.
    /// </summary>
    public interface IRelationType : IMetaObject
    {
        Origin Origin { get; }

        IAssociationType AssociationType { get; }

        IRoleType RoleType { get; }

        Multiplicity Multiplicity { get; }

        bool IsDerived { get; }
    }
}
